client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -544,4 +545,40 @@ object DeltaSharingRestClient extends Logging {
     val value = System.getProperty(key)
     if (value == null) "<unknown>" else value.replace(' ', '_')
   }
+
+  def apply(profileFile: String, forStreaming: Boolean = false): DeltaSharingClient = {
+    val sqlConf = SparkSession.active.sessionState.conf
+
+    val profileProviderclass =
+      sqlConf.getConfString("spark.delta.sharing.profile.provider.class",
+        "io.delta.sharing.client.DeltaSharingFileProfileProvider")
```

Comment:
Can we route this via ConfUtils (below as well) ?

Reply:
moved.